src/resources/views/crud/form_content.blade.php

```diff
@@ -53,11 +53,21 @@ function initializeFieldsWithJavascript(container) {
       });
     }
 
+    function preventUnload(event) {
+      // Cancel the event as stated by the standard.
+      event.preventDefault();
+      // Older browsers supported custom message
+      event.returnValue = '';
+    }
+
     jQuery('document').ready(function($){
 
       // trigger the javascript for all fields that have their js defined in a separate method
       initializeFieldsWithJavascript('form');
 
+      @if($crud->getWarnBeforeLeaving())
+      window.addEventListener('beforeunload', preventUnload);
+      @endif
```

Comment:
```suggestion
    jQuery('document').ready(function($){

      // trigger the javascript for all fields that have their js defined in a separate method
      initializeFieldsWithJavascript('form');

      // retreives the current form data
      const getFormData = () => new URLSearchParams(new FormData(document.querySelector("main form"))).toString();

      // prevents the page unload
      const preventUnload = () => {
        if(initData !== getFormData()) {
          event.preventDefault();
          event.returnValue = "";
        }
      }

      @if($crud->getWarnBeforeLeave())
      let initData = getFormData();
      window.addEventListener('beforeunload', preventUnload);
      @endif
```

Reply:
Hi @jnoordsij!
Can you test this? This loads the initial state of the `form` and at the end, it checks if the form is changed by comparing both strings.

Reply:
Thanks for that, works great for me! I applied this in the last commit.